include/glow/Graph/Graph.h

```diff
@@ -261,10 +270,78 @@ class Function final : public Named {
 
   const Module *getParent() const { return parent_; }
 
+  /// Logs the node creation with a list of input nodes.
+  void logNodeCreation(Node *newNode, llvm::ArrayRef<Node *> inputs) {
+    if (dumpCompilationLogOpt) {
+      std::string logStr = "";
+      logStr.append("[FULL SCOPE: ");
+      logStr.append(logCtx_.getFullScopeName());
+      logStr.append("]");
+      logStr.append(" --- CREATE { (");
+      logStr.append(newNode->getKindName());
+      logStr.append(", ");
+      logStr.append(newNode->getName());
+      logStr.append(") <== ");
+      for (auto n : inputs) {
+        logStr.append(" ( ");
+        logStr.append(n->getKindName());
+        logStr.append(", ");
+        logStr.append(n->getName());
+        logStr.append(" ) ");
+      }
+      logStr.append(" }\n");
+      logCtx_.addLogContent(logStr);
+    }
+  }
+
+  /// Logs the node replacement.
+  void logNodeReplacement(Node *oldNode, Node *newNode) {
+    if (dumpCompilationLogOpt) {
+      std::string logStr = "";
+      logStr.append("[FULL SCOPE: ");
+      logStr.append(logCtx_.getFullScopeName());
+      logStr.append(" ]");
+      logStr.append(" --- REPLACE { ( ");
+      logStr.append(oldNode->getKindName());
+      logStr.append(", ");
+      logStr.append(oldNode->getName());
+      logStr.append(" ) <== ( ");
+      logStr.append(newNode->getKindName());
+      logStr.append(", ");
+      logStr.append(newNode->getName());
+      logStr.append(" ) }\n");
+      logCtx_.addLogContent(logStr);
+    }
+  }
+
+  /// Logs the node deletion.
+  void logNodeDeletion(NodesList::iterator deletedNode) {
+    if (dumpCompilationLogOpt) {
+      std::string logStr = "";
+      logStr.append("[FULL SCOPE: ");
+      logStr.append(logCtx_.getFullScopeName());
+      logStr.append(" ]");
+      logStr.append(" --- DELETE { (");
+      logStr.append(deletedNode->getKindName());
+      logStr.append(", ");
+      logStr.append(deletedNode->getName());
+      logStr.append(") }\n");
+      logCtx_.addLogContent(logStr);
+    }
+  }
+
   /// Inserts the node \p N to the list of nodes, and returns the inserted node.
   template <class NodeTy> NodeTy *addNode(NodeTy *N) {
     N->setName(Module::uniqueName(N->getName(), uniqueNodeNames_));
     nodes_.push_back(N);
+
+    // Log the node creation.
```

Comment:
Why do you need the list of inputs? You can inspect the node N inside the logNodeCreation method.